src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -302,10 +304,10 @@ internal void CloseExtensionControl(IViewExtension viewExtension)
         internal void CloseExtensionTab(object sender, RoutedEventArgs e)
         {
             string tabName = (sender as Button).DataContext.ToString();
-
-            CloseExtension?.Invoke(tabName);
-
             TabItem tabitem = ExtensionTabItems.OfType<TabItem>().SingleOrDefault(n => n.Header.ToString() == tabName);
+
+            var viewExtensionBaseClass = (ViewExtensionBase) tabitem.Tag;
```

Comment:
Thinking about casts like this one. We can't really be sure the extension is going to implement the base class. That means this can fail. What we should do is instead use the `is` or `as` operator and call the `Closed` function conditionally, based on whether the extension implements the base class or not. We should add a test for cases like this as well.

Reply:
I've been liking this syntax lately:
```
 if (tabItem.Tag is ViewExtensionBase base)
        {
            yada yada with base
        }

```

Reply:
Good point. Thanks for catching it. Changed it.

Reply:
Regarding the test for these cases, since we removed the CloseExtension event implementation, I don't think we can add a test for this extensions workflow that do not implement this base class.

Reply:
@reddyashish Whether they implement the base class or not, extensions should still be able to call `CloseExtensionInSideBar`. While that won't result in a call to `Closed`, because it is not implemented, the extension should still be closed.

Reply:
Yeah, we do have a test for that already. 
https://github.com/DynamoDS/Dynamo/blob/2453d84f8628fd5b764011913de5006839f848c7/test/DynamoCoreWpfTests/ViewExtensions/WorkspaceDependencyViewExtensionTests.cs#L118

Reply:
But the workspace dependency extension now implements the base class with this PR. I'm asking for a test where that's not the case.

Reply:
Added it.